test/multi-client.test.js

```diff
@@ -0,0 +1,162 @@
+'use strict';
+
+const { join } = require('path');
+const { spawn } = require('child_process');
+const Client = require('asset-pipe-client');
+const buildServerUri = 'http://127.0.0.1:7200';
+const client = new Client({ buildServerUri });
+const supertest = require('supertest');
+const request = supertest(buildServerUri);
+
+const get = async url => {
+    const { body } = await request
+        .get(url)
+        .set('Accept', 'application/json')
+        .expect('Content-Type', /application\/json/)
+        .expect(200);
+    return body;
+};
+
+const localiseBodyPaths = body =>
+    JSON.parse(
+        JSON.stringify(body, null, 2).replace(
+            /"file":\s".*\/asset-pipe\//g,
+            '"file": "'
+        )
+    );
+
+let server;
+beforeAll(
+    () =>
+        new Promise((resolve, reject) => {
+            server = spawn('./node_modules/.bin/asset-pipe-server', [], {
+                env: Object.assign({}, process.env, {
+                    NODE_ENV: 'development',
+                    PORT: 7200,
+                }),
+            });
+            server.stdout.once('data', () => resolve());
+            server.stderr.once('data', () => resolve());
+            server.once('error', err => reject(err));
+            server.once('close', () => resolve());
+        })
+);
+afterAll(() => {
+    server.kill();
+});
+
+test('Multiple clients upload js feeds to build server', async () => {
+    const js1 = join(__dirname, '..', 'assets', 'a.js');
```

Comment:
just do `join(__dirname, '../assets/a.js')`. The paths are normalized by `path`.

Potentially you can do `require.resolve('../assets/a.js')`. It will throw if the file does not exist, but I don't think that's an issue

Reply:
Nice, replace all with resolve destructured from require which makes it nice and terse. Like it.